test/liquidations.test.ts

```diff
@@ -104,39 +106,295 @@ describe('isBadDebt', () => {
+
+    let [auction, auctionEstimate] = buildAuction(
+      userPositions,
+      result.auctionPercent,
+      result.bid,
+      result.lot,
+      mockPool,
+      mockPoolOracle
+    );
+    let postAuctionUser = userPositions;
+
+    for (let [index, amount] of auction.collateral) {
+      postAuctionUser.collateral.set(index, postAuctionUser.collateral.get(index)! - amount);
+    }
+
+    for (let [index, amount] of auction.liabilities) {
+      postAuctionUser.liabilities.set(index, postAuctionUser.liabilities.get(index)! - amount);
+    }
+
+    const userEstimateAfterAuction = PositionsEstimate.build(
+      mockPool,
+      mockPoolOracle,
+      postAuctionUser
+    );
+
+    expect(result.auctionPercent).toBe(46);
+    expect(result.bid).toContain(USDC);
+    expect(result.lot).toContain(XLM);
+    expect(auctionEstimate.totalSupplied > auctionEstimate.totalBorrowed).toBe(true);
+    expect(userEstimateAfterAuction.totalEffectiveCollateral).toBeGreaterThan(
+      userEstimateAfterAuction.totalEffectiveLiabilities
+    );
+
+    // Check the assets are sorted by value (highest first)
+    const collateralValues = result.lot.map((assetId) => {
+      const index = mockPool.metadata.reserveList.indexOf(assetId);
+      const amount = userPositions.collateral.get(index) || BigInt(0);
+      const price = mockPoolOracle.getPriceFloat(assetId) || 0;
+      const reserve = mockPool.reserves.get(assetId);
+      if (!reserve) return 0;
+      return reserve.toEffectiveAssetFromBTokenFloat(amount) * price;
+    });
+
+    // Check that values are in descending order
+    for (let i = 1; i < collateralValues.length; i++) {
+      expect(collateralValues[i - 1]).toBeGreaterThanOrEqual(collateralValues[i]);
+    }
+
+    const liabilityValues = result.bid.map((assetId) => {
+      const index = mockPool.metadata.reserveList.indexOf(assetId);
+      const amount = userPositions.liabilities.get(index) || BigInt(0);
+      const price = mockPoolOracle.getPriceFloat(assetId) || 0;
+      const reserve = mockPool.reserves.get(assetId);
+      if (!reserve) return 0;
+      return reserve.toEffectiveAssetFromDTokenFloat(amount) * price;
+    });
+
+    // Check that liability values are in descending order
+    for (let i = 1; i < liabilityValues.length; i++) {
+      expect(liabilityValues[i - 1]).toBeGreaterThanOrEqual(liabilityValues[i]);
+    }
   });
 
-  it('should calculate max of 100 percent liquidation size', () => {
-    userEstimate.totalEffectiveCollateral = 1700;
-    userEstimate.totalEffectiveLiabilities = 2200;
-    userEstimate.totalBorrowed = 1900;
-    userEstimate.totalSupplied = 2000;
-    const result = calculateLiquidationPercent(userEstimate);
+  it('should return 100% liquidation when health factor is very poor', () => {
+    // Create positions with very high liabilities compared to collateral
+    userPositions = new Positions(
+      new Map([[USDC_ID, BigInt(500e7)]]),
```

Comment:
this test case might be better with multiple dust collateral positions, to ensure all are included.